### DISCUSSION TITLE
Consistently show/accept UTC dates and times when the preference is set

### DISCUSSION THREAD

**Discussion by HakanL** [datalust/seq-tickets#1107] [Ideas]
My default on my user in SEQ is to use my local timezone, because that's most relevant to reported issues, etc, and with the UTC hover-over it's great to compare to server time (UTC). But there are times where I have a timestamp in UTC (from another log file for example) that I need to find in SEQ. It would be great to have a temporary option to enter date/time in UTC, without having to change my user's time zone.

**Comment by nblumhardt** [comment]
Thanks for the feedback :+1:

It seems worthwhile exploring whether Seq could accept a full ISO-8601 timestamp (with time zone information/Z) pasted into the text portion of the from/to date pickers. May be one option to achieve this with minimal impact on the date selection experience.

**Comment by HakanL** [comment]
That may work, but a lot of times I want a range around a particular timestamp in UTC. But I guess it could transform it to local and then I could add/subtract a few minutes.

**Comment by nblumhardt** [comment]
Another avenue that might work, if you're not doing it too frequently:

```sql
@Timestamp >= DateTime('2018-01-01T12:34Z') and 
@Timestamp < DateTime('2018-01-01T23:45Z')
```

I.e., the `DateTime` function accepts time zone info here.

**Comment by HakanL** [comment]
Yea I prefer to get it into the picker since I then use that as a "master-filter" and add a new criteria in the query string to dig deeper into an issue.

**Comment by HakanL** [comment]
Even a "popup time converter" would work that could be popped open with a little button next to the picker or something, which then would let you select timezone or UTC and it would transfer over to the date/time picker when closed/OKd. It would also help with other timezones (it could have a drop down), even if it's easier to add an hour or two, vs calculating UTC in your head.

**Comment by andymac4182** [comment]
@HakanL not sure it its helpful but the URL has the time in UTC and you can edit it up there.

**Comment by HakanL** [comment]
Yea, that's an option, but not as slick :)

**Comment by nblumhardt** [comment]
Re-titling this, as there are a few other places we should also fix (chart axis labels, in particular).
